### PR TITLE
[RW-885][risk=no] Adding ability to search visit occurrence

### PR DESCRIPTION
**Pull request by osgsusan**
Adding ability to search visit occurrence

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/VisitsQueryBuilder.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        parentList.add(parameter.getConceptId());
+      } else {
+        childList.add(parameter.getConceptId());
+      }
+    }
+
+    // Collect all parent type queries to run them together
+    if (!parentList.isEmpty()) {
+      String namedParameter = "visit" + getUniqueNamedParameterPostfix();
+      queryParams.put(namedParameter, QueryParameterValue.array(parentList.toArray(new Long[0]), Long.class));
```
Comment (freemabd1) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/VisitsQueryBuilder.java]: nit: I would use java 8 here: parentList.toArray(Long[]::new)

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/VisitsQueryBuilder.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      String parentSql = VISIT_SELECT_CLAUSE_TEMPLATE +
+        VISIT_PARENT_CLAUSE_TEMPLATE.replace("${parentIds}", "@" + namedParameter);
+      queryParts.add(parentSql);
+    }
+
+    // Collect all child type queries to run them together
+    if (!childList.isEmpty()) {
+      String namedParameter = "visit" + getUniqueNamedParameterPostfix();
+      queryParams.put(namedParameter, QueryParameterValue.array(childList.toArray(new Long[0]), Long.class));
```
Comment (freemabd1) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/VisitsQueryBuilder.java]: nit: I would use java 8 here: parentList.toArray(Long[]::new)

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/VisitsQueryBuilder.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    List<String> queryParts = new ArrayList<>();
+    Map<String, QueryParameterValue> queryParams = new HashMap<>();
+
+    List<Long> parentList = new ArrayList<>();
+    List<Long> childList = new ArrayList<>();
+    for (SearchParameter parameter : inputParameters.getParameters()) {
+      if (parameter.getGroup()) {
+        parentList.add(parameter.getConceptId());
+      } else {
+        childList.add(parameter.getConceptId());
```
Comment (freemabd1) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/VisitsQueryBuilder.java]: Should we have validation of some sort to check that conceptId is not null? And maybe throw a BadRequestException? If so, then maybe we should add a test case as well?

Comment (osgsusan) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/VisitsQueryBuilder.java]: We should be consistent, sure. done.

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/VisitsQueryBuilder.java (hunk 1 above)
Comment (osgsusan) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/VisitsQueryBuilder.java]: done

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/VisitsQueryBuilder.java (hunk 2 above)
Comment (osgsusan) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/VisitsQueryBuilder.java]: done

**Pull request merged**